lib/reactotron-react-native/src/helpers/parseURL.test.ts

```diff
@@ -0,0 +1,59 @@
+import { getHostFromUrl } from "./parseURL"
+
+describe("getHostFromUrl", () => {
+  it("should throw when no host is found", () => {
+    expect(() => {
+      getHostFromUrl("")
+    }).toThrow()
+  })
+
+  it("should get host from URL without scheme", () => {
+    Object.entries({
+      localhost: "localhost",
+      "127.0.0.1": "127.0.0.1",
+    }).forEach(([host, url]) => {
+      expect(getHostFromUrl(url)).toEqual(host)
+    })
+    expect(getHostFromUrl("localhost")).toEqual("localhost")
+    expect(getHostFromUrl("127.0.0.1")).toEqual("127.0.0.1")
+  })
+
+  it("should get the host from URL with http scheme", () => {
+    Object.entries({
+      localhost: "http://localhost",
+      "example.com": "http://example.com",
+    }).forEach(([host, url]) => {
+      expect(getHostFromUrl(url)).toEqual(host)
+    })
+  })
+
+  it("should get the host from URL with https scheme", () => {
+    Object.entries({
+      localhost: "https://localhost",
+      "example.com": "https://example.com",
+    }).forEach(([host, url]) => {
+      expect(getHostFromUrl(url)).toEqual(host)
+    })
+  })
+
+  it("should get the host from URL and ignore path, port, and query params", () => {
```

Comment:
Should we also test with Expo tunnel-style URLs? https://docs.expo.dev/more/expo-cli/#tunneling.

Reply:
I'm not quite sure how to support tunneling honestly. I think you'd have to create a tunnel for each port that you wanted to connect to. This is something I'd probably reach for Tailscale for... which reminds me I should create a blog post for that. Either way, I think we'd still get the host right for it. Just would have to make sure we forwarded the right ports to Reactotron, yes?

Reply:
With this comment I only meant we should check we don't break anything that wasn't broken for Expo's tunneling support. I'm not sure if `exp://` is used there for hosting the JS bundle, I thought maybe it was just used for deep links now? Not sure.

Reply:
It appears to work with Expo Go without it. Removed.

Reply:
@lindboe In order to support tunneling, I believe we'd have to have some way to know to include the port. This PR only replaces the old method of finding the host with safeguards around not crashing the application if the URL is not valid. Maybe we tackle that in another PR?

Reply:
Yeah that makes sense 👍